lms/djangoapps/dashboard/sysadmin.py

```diff
@@ -624,8 +626,16 @@ def get(self, request, *args, **kwargs):
             try:
                 course = get_course_by_id(course_id)
             except Exception:
-                log.info('Cannot find course %s', course_id)
-                raise Http404
+                course_key = CourseLocator(
```

Comment:
Instead of nesting these exceptions, could you coerce the course_id to the CourseLocator object around line 627 and then simply make the call please.